api/api-pipeline-config-v6/src/main/java/com/thoughtworks/go/apiv6/admin/pipelineconfig/representers/PipelineConfigRepresenter.java

```diff
@@ -0,0 +1,199 @@
+import com.thoughtworks.go.apiv6.admin.pipelineconfig.representers.trackingtool.TrackingToolRepresenter;
+import com.thoughtworks.go.apiv6.shared.representers.EnvironmentVariableRepresenter;
+import com.thoughtworks.go.apiv6.shared.representers.configorigin.ConfigRepoOriginRepresenter;
+import com.thoughtworks.go.apiv6.shared.representers.configorigin.ConfigXmlOriginRepresenter;
+import com.thoughtworks.go.apiv6.shared.representers.stages.StageRepresenter;
+import com.thoughtworks.go.config.*;
+import com.thoughtworks.go.config.exceptions.UnprocessableEntityException;
+import com.thoughtworks.go.config.materials.MaterialConfigs;
+import com.thoughtworks.go.config.remote.ConfigOrigin;
+import com.thoughtworks.go.config.remote.FileConfigOrigin;
+import com.thoughtworks.go.config.remote.RepoConfigOrigin;
+import com.thoughtworks.go.spark.Routes;
+import org.bouncycastle.crypto.InvalidCipherTextException;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Consumer;
+
+public class PipelineConfigRepresenter {
+
+    public static void toJSON(OutputWriter jsonWriter, PipelineConfig pipelineConfig) {
+        jsonWriter.addLinks(linksWriter -> linksWriter
+                .addLink("self", Routes.PipelineConfig.name(pipelineConfig.getName().toString()))
+                .addAbsoluteLink("doc", Routes.PipelineConfig.DOC)
+                .addLink("find", Routes.PipelineConfig.find()));
+        // This is needed for the case when there are no materials defined. Refer to pipeline_config_representer.rb#152
+        pipelineConfig.errors().addAll(pipelineConfig.materialConfigs().errors());
+        if (!pipelineConfig.errors().isEmpty()) {
+            jsonWriter.addChild("errors", errorWriter -> {
+                HashMap<String, String> errorMapping = new HashMap<>();
+                errorMapping.put("labelTemplate", "label_template");
+                errorMapping.put("params", "parameters");
+                errorMapping.put("variables", "environment_variables");
+                errorMapping.put("trackingTool", "tracking_tool");
+
+                new ErrorGetter(errorMapping).toJSON(errorWriter, pipelineConfig);
+            });
+        }
+
+        jsonWriter.add("label_template", pipelineConfig.getLabelTemplate());
+        jsonWriter.add("lock_behavior", pipelineConfig.getLockBehavior());
+        jsonWriter.add("name", pipelineConfig.name());
+        jsonWriter.add("template", pipelineConfig.getTemplateName());
+        writeOrigin(jsonWriter, pipelineConfig.getOrigin());
+        jsonWriter.addChildList("parameters", getParams(pipelineConfig.getParams()));
+        jsonWriter.addChildList("environment_variables", envVarsWriter -> EnvironmentVariableRepresenter.toJSON(envVarsWriter, pipelineConfig.getVariables()));
+        jsonWriter.addChildList("materials", getMaterials(pipelineConfig.materialConfigs()));
+        if (pipelineConfig.getStages().isEmpty()) {
```

Comment:
Why do we need to make stages null if it's not present, rather than leaving it as an empty list?